chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/edition2015/TestFunctionalityAllowedByRoleReviewer.java

```diff
@@ -29,49 +28,35 @@
 
     private FF4j ff4j;
     private Environment env;
-    private ErrorMessageUtil errorMessages;
-    private ResourcePermissions permissions;
+    private List<RestrictedCriteriaTestFunctionality> restrictedCriteriaTestFunctionality;
 
     @Autowired
-    public TestFunctionalityAllowedByRoleReviewer(FF4j ff4j, Environment env, ResourcePermissions permissions,
-            ErrorMessageUtil errorMessages) {
+    public TestFunctionalityAllowedByRoleReviewer(FF4j ff4j, Environment env) {
         this.ff4j = ff4j;
         this.env = env;
-        this.errorMessages = errorMessages;
-        this.permissions = permissions;
+    }
+
+    @PostConstruct
+    public void setup() {
+        restrictedCriteriaTestFunctionality = getRestrictedCriteriaTestFunctionality();
     }
 
     @Override
     public void review(PendingCertifiedProductDTO listing) {
         if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE_PLUS_ONE_WEEK)) {
             for (PendingCertificationResultDTO cr : listing.getCertificationCriterion()) {
                 if (cr.getTestFunctionality() != null) {
-                    for (PendingCertificationResultTestFunctionalityDTO crtf : cr.getTestFunctionality()) {
-                        if (!isTestFunctionalityChangeAllowedBasedOnRole(cr.getCriterion().getId(),
-                                crtf.getTestFunctionalityId())) {
-                            listing.getErrorMessages()
-                                    .add(errorMessages.getMessage("listing.criteria.testFunctionalityPermissionError",
-                                            crtf.getNumber(), cr.getCriterion().getNumber()));
-                        }
-                    }
+                    cr.getTestFunctionality().removeIf(
+                            crtf -> findRestrictedTestFunctionality(
+                                    cr.getCriterion().getId(),
+                                    crtf.getTestFunctionalityId()).isPresent());
                 }
-
             }
         }
     }
 
-    private boolean isTestFunctionalityChangeAllowedBasedOnRole(Long criteriaId, Long testFunctionalityId) {
```

Comment:
Does removing this mean that there are no ROLE based requirements around TF stuff? I thought this was one of those places where ACB was restricted, but ONC/ADMIN wasn't